### PR TITLE
Fix test failures on main

### PR DESCRIPTION
**Pull request by andreww**
This fixes two (of the three) failing tests on main.

The issues that are fixed involves a test case where newlines are printed by cats, which are not in the reference output in the tests (an easy fix).

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -39,6 +39,11 @@ jobs:
         flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
         # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
         flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+    - name: Check types with mypy
+      run: |
+        python3 -m pip install mypy
```
Comment (abhidg) [inline .github/workflows/tests.yml:44]: Ideally these should be in test dependencies and be installable in one line

Comment (andreww) [reply .github/workflows/tests.yml:44]: You mean inside `project.optional-dependencies` in pyproject.toml?

Comment (andreww) [reply .github/workflows/tests.yml]: I see - done this now.

**Comment by andreww** [comment]
I think some of my "fixes" in here are properly addressed in #80, 058a780 in particular. @tlestang - we should probably figure out how to order these two PRs, or if it's best to chop this one up / just take the stuff that's not involved in #80.

**Comment by tlestang** [comment]
Hi @andreww  - I split the rewriting/testing of the carbon footprint estimation module into two PR #79 and #80. I just merged #80 in which fixed the failing test on `main`. I also resolved a few inconsistencies (unused function or variable like `jobinfo`) that were necessary to introduce in order to integrate the changes in two distinct steps without breaking `cats`.

File: cats/configure.py [hunk 2]
```diff
@@ -24,7 +24,8 @@
 __all__ = ["get_runtime_config"]
 
 
-def get_runtime_config(args) -> tuple[dict, APIInterface, str, int]:
+def get_runtime_config(args) -> tuple[Mapping[str, Any], APIInterface, str, int, 
+                                      Union[list[tuple[int, float]],None],Any]:
```
Comment (tlestang) [inline cats/configure.py:28]: one of these cases where typing really isn't worth the trouble in my opinion.

**Comment by tlestang** [comment]
About the conflicts:
- The `check_clean_arguments.py` was removed, because the job of processing and validating command line arguments is now carried out in the config module.
- After #80 the in-memory representation of the configuration file (`config` dictionary) doesn't have to leave the config module, so the `config.get_runtime_config` function doesn't have to return it in `__init__.py`. Same goes for `_jobinfo`.

**Comment by andreww** [comment]
Thanks - I'll reorder and rebase this branch so that it makes sense now #80 is merged.

**Comment by andreww** [comment]
Rebased and cleaned all of this up. Two new PRs inbound. We can kill this one.

**Pull request closed without merging**